Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -32,9 +32,47 @@ public struct Configuration: Equatable {
     public let rules: [Rule]
     public var rootPath: String?              // the root path to search for nested configurations
     public var configurationPath: String?     // if successfully loaded from a path
-    public var hash: Int?
     public let cachePath: String?
 
+    internal var cacheDescription: String {
+        let cacheRulesDescriptions: [String: Any] = rules.reduce([:]) { accu, element in
+            var accu = accu
+            accu[type(of: element).description.identifier] = element.cacheDescription
+            return accu
+        }
+        let dict: [String: Any] = [
+            "root": rootPath ?? FileManager.default.currentDirectoryPath,
+            "rules": cacheRulesDescriptions
+        ]
+        if let jsonData = try? JSONSerialization.data(withJSONObject: dict),
```

Comment:
> But as I tried it now, the order of Set does not seem to be constant by Yams.

Ah, both of `JSONSerialization ` and `Yams` do not serialize `Set` directly and can't serialize `Set`.
They are using results of inserting `Set` into String interpolation.
Inserting `Set` into String interpolation in `public var consoleDescription: String {…}` seems non-deterministic on Linux. 🤔